ext/standard/pack.c

```diff
@@ -750,7 +750,16 @@ PHP_FUNCTION(unpack)
 			c = *format;
 
 			if (c >= '0' && c <= '9') {
-				repetitions = atoi(format);
+				errno = 0;
+				long tmp = strtol(format, NULL, 10);
+				/* There is not strtoi. We have to check the range ourselves.
+				 * On 32-bit the INT_{MIN,MAX} are useless because long == int, but on 64-bit they do limit us to 32-bit. */
+				if (errno || tmp < INT_MIN || tmp > INT_MAX) {
+					php_error_docref(NULL, E_WARNING, "Type %c: integer overflow", type);
+					zend_array_destroy(Z_ARR_P(return_value));
+					RETURN_FALSE;
+				}
```

Comment:
I suppose you double-checked that we don't have something that does something like a ``strtoi`` in ``zend_operators.h`` ?

But otherwise this looks good to me. :)

Reply:
We have a `ZEND_STRTOL` which depending on the size of a long does strtoll for 64-bit or strtol for 32-bit.
But since we don't care about the fact that it can return long (we only want an int anyway), this seemed very confusing especially since the macro name doesn't always match the functionality.

Reply:
ACK.